src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php

```diff
@@ -97,43 +94,28 @@ public function handlePersist(PersistEvent $event)
      */
     public function handleReorder(ReorderEvent $event)
     {
-        $node = $event->getNode();
         $document = $event->getDocument();
 
-        if (false == $this->supports($document)) {
+        if (!$document instanceof OrderBehavior) {
             return;
         }
 
-        $propertyName = $this->encoder->systemName(self::FIELD);
-
-        $parent = $node->getParent();
-        $count = 0;
-        foreach ($parent->getNodes() as $childNode) {
-            $childNode->setProperty($propertyName, ($count + 1) * 10, PropertyType::LONG);
-            ++$count;
-        }
-
-        $this->handleHydrate($event);
-    }
+        $parentDocument = $this->inspector->getParent($document);
 
-    /**
-     * Adds the order to the document.
-     *
-     * @param AbstractMappingEvent $event
-     */
-    public function handleHydrate(AbstractMappingEvent $event)
-    {
-        if (false == $this->supports($event->getDocument())) {
+        if (null === $parentDocument) {
             return;
         }
 
-        $node = $event->getNode();
-
-        $order = $node->getPropertyValueWithDefault(
-            $this->encoder->systemName(self::FIELD),
-            null
-        );
-
-        $event->getAccessor()->set('suluOrder', $order);
+        $count = 0;
+        foreach ($this->inspector->getChildren($parentDocument) as $childDocument) {
```

Comment:
@danrot have refactored the reordering to use the documents instead of the PHPCR nodes, as we should avoid talking to PHPCR nodes directly and the writing to nodes is now done automatically because of the Metadata mapping.

This has the effect that only documents implementing OrderBehavior will have the order assigned to them.